jsettlers.logic/src/main/java/jsettlers/logic/map/grid/objects/MapObjectsManager.java

```diff
@@ -400,18 +401,14 @@ public boolean pushMaterial(short x, short y, EMaterialType materialType) {
 		}
 	}
 
-	public ShortPoint2D pushMaterialForced(short x, short y, EMaterialType materialType) {
-		IMapArea mapArea = new HexGridArea(x, y, 0, 200);
-
-		for (ShortPoint2D position : mapArea) {
-			short currX = position.x;
-			short currY = position.y;
+	public ShortPoint2D pushMaterialForced(int x, int y, EMaterialType materialType) {
+		return HexGridArea.iterate(x, y, 0, 200, (currX, currY, radius) -> {
 			if (grid.isInBounds(currX, currY) && canForcePushMaterial(currX, currY, materialType)) {
 				pushMaterial(currX, currY, materialType);
-				return position;
+				return Optional.of(new ShortPoint2D(currX, currY));
 			}
-		}
-		return null;
+			return Optional.empty();
+		}).orElseGet(() -> null);
```

Comment:
You should avoid `orElse(null)` whenever possible. This is a code smell.

Evaluate the Optional as late as possible. (Except you recover the absence of the value in the middle with a really good recover value)
If your Interfaces provide Optional instead of the possibility of null you can omit the null checks without the runtime possibility of NullPointerExceptions.
I try to show you what I mean with the following two code examples:

```
class Foo {

        public static void main(String[] args) {
                System.out.println("Hallo");
                System.out.println(third());
        }

        public static Integer third() {
                Integer secondResult = second();
                if (secondResult == null)
                        return null;
                return secondResult * 2;
        }

        public static Integer second() {
                Integer firstResult = first();
                if (firstResult == null)
                        return null;
        return firstResult + 15;

        }

        public static Integer first() {
                return null;
        }
}
```

```
import java.util.Optional;

class Bar {

        public static void main(String[] args) {
                System.out.println("Hallo");
                // Here you can decide what to do if this is empty. Recover it? End the program? Ignore it. What you like
                System.out.println(third().orElse(42));
        }

        public static Optional<Integer> third() {
                return second().map(second -> second * 2);
        }

        public static Optional<Integer> second() {
                return first().map(first -> first + 15);
        }

        public static Optional<Integer> first() {
                return Optional.empty();
        }
}
```

I can imagine only two exceptions when using `orElse(null)` is allowed:
1) If you provide code which is used by other code which is not under your control thus you can't change the interface to Optional<T>
2) If you refactor step by step and seek for small commits your can introduce Optional<T> step by step by using `orElse(null)` in the meantime. But your goal at the end should be to avoid `orElse(null)`.